registry/src/main/java/de/unistuttgart/iaas/amyassist/amy/registry/LocationRegistryImpl.java

```diff
@@ -85,40 +70,19 @@ public void save(Location location) {
         super.save(location);
     }
 
-    /**
-     * Get at most one entity where the attribute is set to true
-     * @param attribute
-     * @return an entity of null
-     */
-    private LocationImpl getEntityWithBooleanAttribute(String attribute) {
-        CriteriaBuilder builder = entityManager.getCriteriaBuilder();
-        CriteriaQuery<LocationImpl> query = builder.createQuery(LocationImpl.class);
-        Root<LocationImpl> root = query.from(LocationImpl.class);
-        query.select(root).where(builder.equal(root.get(attribute), true));
-
-        TypedQuery<LocationImpl> typedQuery = entityManager.createQuery(query);
-
-        List<LocationImpl> homes = typedQuery.getResultList();
-        if (!homes.isEmpty()) {
-            return homes.get(0);
-        } else {
-            return null;
-        }
-    }
-
     /**
      * Get the home location
      * @return entity or null
      */
-    public LocationImpl getHome() {
-        return getEntityWithBooleanAttribute("isHome");
+    public Location getHome() {
+        return getEntityWithTag(AbstractTaggableRegistry.HOME);
```

Comment:
what if there are two locations tagged `home`

Reply:
a RegistryException will be thrown. I can't think of a better way to handle this situation.